frame/referenda/src/lib.rs

```diff
@@ -1270,4 +1280,140 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Self::deposit_event(Event::<T, I>::MetadataCleared { index, hash });
 		}
 	}
+
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// The following assertions must always apply.
+	///
+	/// General assertions:
+	///
+	/// * `ReferendumCount` must always be equal to the number of referenda in `ReferendumInfoFor`.
+	/// * Referendum indices must be sorted in the `ReferendumInfoOf` storage map.
+	/// * Referendum indices in `MetadataOf` must also be stored in `ReferendumInfoOf`.
+	///
+	/// Looking at referenda info:
+	///
+	/// - Data regarding ongoing phase:
+	///
+	/// * The submission deposit cannot be less than `T::SubmissionDeposit`.
+	/// * There must exist track info for the track of the referendum.
+	/// * The decision deposit if provided cannot be less than the decision deposit of the track.
+	/// * The deciding stage has to begin before confirmation period.
+	///
+	/// - Data in `ReferendumInfo::Approved`, `ReferendumInfo::Rejected`,
+	///   `ReferendumInfo::Cancelled` and `ReferendumInfo::TimedOut`:
+	///
+	/// * The submission deposit cannot be less than `T::SubmissionDeposit`.
+	///
+	/// Looking at tracks:
+	/// * The `TrackQueue` should be empty if `DecidingCount` is less than
+	///   `TrackInfo::max_deciding`.
+	/// * The referendum indices stored in TrackQueue` must exist as keys in the `ReferendumInfoFor`
+	///  storage map.
+	#[cfg(any(feature = "try-runtime", test))]
+	fn do_try_state() -> DispatchResult {
+		use sp_runtime::traits::Bounded;
+
+		ensure!(
+			ReferendumCount::<T, I>::get() as usize ==
+				ReferendumInfoFor::<T, I>::iter_keys().count(),
+			DispatchError::Other(
+				"Number of referenda in `ReferendumInfoFor` is greater than `ReferendumCount`"
+			)
+		);
+
+		// The order of indices when running `queueing_works` is: 0, 4, 2, 1, 3 🤔 
+		/*
+		ensure!(
+			ReferendumInfoFor::<T, I>::iter_keys()
```

Comment:
Any idea why is this happening?

Reply:
probably this does not return in order it was added, the StorageMap is a tree